tests/jballerina-unit-test/src/test/java/org/ballerinalang/nativeimpl/jvm/runtime/api/tests/Values.java

```diff
@@ -500,10 +500,7 @@ public static BError getErrorValue(BString errorTypeName) {
         BString errorMsg = StringUtils.fromString("error message!");
         BError bError = ErrorCreator.createError(errorModule, errorTypeName.getValue(), errorTypeName,
                 ErrorCreator.createError(errorMsg), ValueCreator.createMapValue());
-        // TODO: fix https://github.com/ballerina-platform/ballerina-lang/issues/41025
-        String typeName = errorTypeName.getValue().equals("ResourceDispatchingError") ?
-                "RequestDispatchingError" : errorTypeName.getValue();
-        assert bError.getType().getName().equals(typeName);
+        assert bError.getType().getName().equals(errorTypeName.getValue());
```

Comment:
As I know these assertions do not work normally and we need to pass a flag to enable them. So it be better to move this to a `if` condition and throw an assertion error.